ToolBox/SECDataDownloader/SECDataDownloader.cs

```diff
@@ -319,24 +339,30 @@ public void Download(string rawDestination, DateTime start, DateTime end)
                             _indexGate.WaitToProceed();
 
                             Log.Trace("SECDataDownloader.Download(): Downloading CIK lookup data");
-                            var cikLookupBytes = client.GetByteArrayAsync($"{BaseUrl}/cik-lookup-data.txt")
+                            var cikLookupBytes = _client.GetByteArrayAsync($"{BaseUrl}/cik-lookup-data.txt")
                                 .SynchronouslyAwaitTaskResult();
-                            
+
                             File.WriteAllBytes(cikLookupTempPath, cikLookupBytes);
                             File.Move(cikLookupTempPath, cikLookupPath);
                             File.Delete(cikLookupTempPath);
                         }
                     }
-                }
-                catch (HttpRequestException err)
-                {
-                    if (err.StatusCode == HttpStatusCode.Forbidden || err.StatusCode == HttpStatusCode.TooManyRequests)
+                    catch (HttpRequestException err)
                     {
-                        Log.Trace($"SECDataDownloader.Download(): Rate limited downloading CIK-ticker mappings - retrying in 10s");
-                        Thread.Sleep(10000);
+                        if (err.StatusCode == HttpStatusCode.Forbidden ||
+                            err.StatusCode == HttpStatusCode.TooManyRequests)
+                        {
+                            Log.Trace(
+                                $"SECDataDownloader.Download(): Rate limited downloading CIK-ticker mappings - retrying in 10s");
+                            Thread.Sleep(10000);
+                        }
                     }
                 }
             }
+            finally
+            {
+                _client.DisposeSafely();
```

Comment:
If we are going to dispose of the client at the end of `Download()` we should create a new one per each call at the top or we should make the class IDiposable and dispose on the client there, else two consecutive calls would break. Think this means we are not expecting consecutive calls most of the cases so we could just go for the first approach of creating the client at the top of the method 👍